tests/unit/mixins/route-test.js

```diff
@@ -440,3 +440,83 @@ test('it allows overrides/manual invocations of updateInfinityModel', assert =>
   assert.equal(model.get('content.length'), 3);
   assert.equal(model.get('content.lastObject.title'), 'Tender Is the Night', 'updateInfinityModel can be invoked manually');
 });
+
+/*
+ * Compatibility Tests
+ */
+var dummyStore = {
+  _dummyFetch(modelType, findQuery) {
+    var items = [
+      { id: 1, title: 'The Great Gatsby' },
+      { id: 2, title: 'The Last Tycoon' }
+    ];
+    var item = items[findQuery.page-1];
+    return new Ember.RSVP.Promise(resolve => {
+      Ember.run(this, resolve, Ember.ArrayProxy.create({
+        content: Ember.A([item]),
+        meta: { total_pages: 2 }
+      }));
+    });
+  },
+  query(modelType, findQuery) {
+    return this._dummyFetch(modelType, findQuery);
+  },
+  find(modelType, findQuery) {
+    return this._dummyFetch(modelType, findQuery);
+  }
+};
+
+test('It uses Query for ED >= 1.13.4', assert => {
+
+  var RouteObject = Ember.Route.extend(RouteMixin, {
+    store: dummyStore,
+    model() {
+      return this.infinityModel('item', { perPage: 1 });
+    }
+  });
+  
+  var route = RouteObject.create();
+
+  DS.VERSION = "1.13.4";
+  return route.model().then(function() {
+    assert.equal(route.get('_storeFindMethod'), 'query');
+  });
+});
+
+test('It uses Find for ED <= 1.0.0-beta.19.2', assert => {
+  var RouteObject = Ember.Route.extend(RouteMixin, {
+    store: dummyStore,
+    model() {
+      return this.infinityModel('item', { perPage: 1 });
+    }
+  });
+  
+  var route = RouteObject.create();
+
+  DS.VERSION = "1.0.0-beta.19.2";
+  return route.model().then(function() {
+    assert.equal(route.get('_storeFindMethod'), 'find');
+  });
+});
+
+test('It explodes when using an unsupported ED', assert => {
+  var RouteObject = Ember.Route.extend(RouteMixin, {
+    store: dummyStore,
+    model() {
+      return this.infinityModel('item', { perPage: 1 });
+    }
+  });
+  
+  var route = RouteObject.create();
+
+  DS.VERSION = "1.0.0-beta.19.3";
+  
+  var infinityError;
+  try {
+    route.model();
+  } catch(error) {
+    infinityError = error;
+  }
+  
+  assert.equal(infinityError.message, "Ember Infinity: You are using an unsupported version of Ember Data.  Please upgrade to at least 1.13.4 or downgrade to 1.0.0-beta.19.2");
```

Comment:
Consider using `QUnit.throws()` ( https://api.qunitjs.com/throws/ ) for this test

```js
QUnit.throws(() => {
  route.model();
}, /unsupported version of Ember Data/);
```